src/motd.rs

```diff
@@ -25,27 +28,70 @@ impl RegexMatches {
     }
 }
 
+pub struct Counters {
+    tcp: Stats,
+    udp: Stats,
+}
+
+impl Counters {
+    pub fn new(tcp: Stats, udp: Stats) -> Self {
+        Counters {
+            tcp,
+            udp,
+        }
+    }
+}
+
 pub struct Motd {
     start_date_regex: RegexMatches,
     uptime_regex: RegexMatches,
+    tcp_in_regex: RegexMatches,
+    tcp_out_regex: RegexMatches,
+    udp_in_regex: RegexMatches,
+    udp_out_regex: RegexMatches,
     start_date: DateTime<Local>,
+    counters: Counters,
     template: String,
 }
 
 impl Motd {
-    pub fn new(template: String) -> Motd {
+    pub fn new(template: String, counters: Counters) -> Motd {
         let start_date_regex = Regex::new(r"(?i)\{\{\s*start_date\s*\}\}")
             .expect("Failed to compile start_date regex");
         let uptime_regex = Regex::new(r"(?i)\{\{\s*uptime\s*\}\}")
             .expect("Failed to compile uptime regex");
+        let tcp_in_regex = Regex::new(r"(?i)\{\{\s*tcp_in\s*\}\}")
+            .expect("Failed to compile tcp_in regex");
+        let tcp_out_regex = Regex::new(r"(?i)\{\{\s*tcp_out\s*\}\}")
+            .expect("Failed to compile tcp_out regex");
+        let udp_in_regex = Regex::new(r"(?i)\{\{\s*udp_in\s*\}\}")
+            .expect("Failed to compile udp_in regex");
+        let udp_out_regex = Regex::new(r"(?i)\{\{\s*udp_out\s*\}\}")
+            .expect("Failed to compile udp_out regex");
         Motd {
             start_date_regex: RegexMatches::new(&template, start_date_regex),
             uptime_regex: RegexMatches::new(&template, uptime_regex),
+            tcp_in_regex: RegexMatches::new(&template, tcp_in_regex),
+            tcp_out_regex: RegexMatches::new(&template, tcp_out_regex),
+            udp_in_regex: RegexMatches::new(&template, udp_in_regex),
+            udp_out_regex: RegexMatches::new(&template, udp_out_regex),
             start_date: Local::now(),
+            counters,
             template,
         }
     }
 
+    fn summary(source: u64) -> String {
+        match source {
+            0..=999 => format!("{}",source),
+            1000..=999999 => format!("{0:.1}K", source as f32 / 1000.0),
+            1000000..=999999999 => format!("{0:.1}M", source as f32 / 1000000.0),
+            1000000000..=999999999999 => format!("{0:.1}G", source as f32 / 1000000000.0),
+            1000000000000..=u64::MAX => format!("{0:.1}T", source as f32 / 1000000000000.0),
+            _ => panic!("Packet counters have values impossible")
```

Comment:
`unreachable!("Packets counter has impossible value")`

Reply:
Do we really need this branch at all?

Reply:
It seems yes: https://github.com/rust-lang/rust/issues/50907